packages/resolver/src/adapters/cns/fetcher.ts

```diff
@@ -0,0 +1,78 @@
+import {CNSMetadata, CNSUserRecord} from './types'
+import {hexToString} from './utils'
+import {fetchData, handleApiError, isLeft} from '@yoroi/common'
+import {AxiosRequestConfig} from 'axios'
+
+export const getAssetAddress = async (
```

Comment:
build a wrapper for all these functions `handleCns...` -> that returns these functions, {getAsset.., getMetadata...}, it should receive the `fecthData` as a dependency along with everything that is pre-requisite and static while the module is living, I.e -> baseUrl